python/tk_desktop/desktop_engine_project_implementation.py

```diff
@@ -131,6 +134,13 @@ def _register_commands(self):
                 time.sleep(0.5)
 
             self._project_comm.call("trigger_register_command", name, gui_properties, groups)
+            project_command = {"name": name}
+            project_command.update(gui_properties)
+            project_commands.append(project_command)
+
+        # Let the proxy know command registration is complete and which
+        # commands were registered for this project.
+        self._project_comm.call_no_response("project_commands_finished", project_commands)
```

Comment:
Also, you will need to add code that either allows to catch if an RPC call doesn't exist, as you are not guaranteed a client has the latest site engine version. You can 't simply try/except around `call_no_response`either, as you could get an exception because the server died or because the server performed some invalid operation. I would recommend to introduce a new exception type and raise it [here](https://github.com/shotgunsoftware/tk-desktop/blob/master/python/tk_desktop/rpc.py#L149). I think custom types pass through the RPC call, so the other class should be accessible from the caller side as well. At the minimum, you can test the exception's `__str__` to know if it 's an unknown call.